pkg/combination/combination_test.go

```diff
@@ -35,18 +35,40 @@ var combinationTests = []struct {
 			err:          nil,
 		},
 	},
+	{
+		name:  "standard set of long args",
+		input: testdata.LongCombinationInput,
+		expected: expected{
+			combinations: testdata.LongCombinationOutput,
+			err:          nil,
+		},
+	},
 }
 
 func TestAll(t *testing.T) {
 	for _, tt := range combinationTests {
 		t.Run(tt.name, func(t *testing.T) {
 			combinationStream := NewStream(
 				WithArgs(tt.input),
-				WithSolveAhead(),
+				WithSolveAhead(true),
```

Comment:
Since we're remove the `cs.All()` method, we could rename this function to be `TestNextAll()` and the `TestNext()` that currently exists to be `TestNextIterative`. Though I would prefer that both cases be covered by the same test. I.E, just one single `TestNext` that tests both of the combination solution methods.

Reply:
Implemented one singular TestNext() that combines the two tests.